### PR TITLE
fix:add check for subscriptions when wrapping resolvers

### PR DESCRIPTION
**Pull request by StephenCoady**
Fixes: https://issues.jboss.org/browse/AEROGEAR-8438

### CODE REVIEW

**Comment by wtrocki** [comment]
Unit test will be handy for this case,  but please do not feel that it is required.

**Comment by darahayes** [comment]
+1 some unit tests here would be really nice.

* The happy case with no subscriptions.
* The case with a mixture of subscriptions and regular resolvers

This is a core function which has 0 code coverage right now. I know it will take you some extra time @StephenCoady but it would be a massive help!

**Comment by wtrocki** [comment]
@StephenCoady Rebase on top of master and force push

File: packages/apollo-voyager-tools/src/wrapResolvers.test.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    return (obj, args, context, info) => {
+      return context
+    }
+  }
+}
+
+const resolversWithOutSubscription = {
+  Query: {
+    allTasks: async (context: any) => {
+      return context.db.select().from('tasks')
```
Comment (darahayes) [inline packages/apollo-voyager-tools/src/wrapResolvers.test.ts]: It's kinda weird to see this code in our unit tests. Let's have a chat about how to test this stuff properly

File: packages/apollo-voyager-tools/src/wrapResolvers.test.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  }
+}
+
+test('Without subscriptions', (t) => {
+  const config: any = {
+    auditLogger,
+    metrics,
+    securityService: {} as any
+  }
+  const result = wrapResolvers(voyagerResolvers(resolversWithOutSubscription, config), resolverFunction())
```
Comment (darahayes) [inline packages/apollo-voyager-tools/src/wrapResolvers.test.ts]: So ideally, we want to test that our resolver functions were correctly wrapped with the wrapper function provided and that when we now execute our resolver, both the wrapper and the original resolver were both called. Can you think of any ideas about how we might do that? 🙂

File: packages/apollo-voyager-tools/package.json [hunk 3]
```diff
@@ -21,6 +21,9 @@
     "graphql-tools": "^4.0.3"
   },
   "devDependencies": {
+    "@aerogear/apollo-voyager-audit": "0.2.7-rc3",
```
Comment (darahayes) [inline packages/apollo-voyager-tools/package.json]: Just a minor thing: make sure this and the metrics ones are removed.

Comment (StephenCoady) [reply packages/apollo-voyager-tools/package.json]: ahh, never saved the file. 2 secs

File: packages/apollo-voyager-tools/package.json [hunk 4]
```diff
@@ -21,6 +21,7 @@
     "graphql-tools": "^4.0.3"
   },
   "devDependencies": {
+    "@aerogear/apollo-voyager-server": "0.2.7-rc3",
```
Comment (wtrocki) [inline packages/apollo-voyager-tools/package.json]: This is circular dependency. No way this is going to work. 
We got lucky with the build.
Disabled that and logged separate issue #80

Comment (darahayes) [reply packages/apollo-voyager-tools/package.json]: @wtrocki this is only a dev dependency used in the tests to pull in some interface needed to allow compilation. It's not required in the production dependencies.

Comment (wtrocki) [reply packages/apollo-voyager-tools/package.json]: Doesn't matter - when you will end up with clean project without stuff compiled typescript compiler will be forced to resolve `chicken or egg problem`.

**Pull request merged**